### PR TITLE
Modify default method for address and country

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**

### CODE REVIEW

File: backend/app/controllers/spree/admin/stock_locations_controller.rb [hunk 1]
```diff
@@ -7,9 +7,10 @@ class StockLocationsController < ResourceController
 
       def set_country
         @stock_location.country = Spree::Country.default
-      rescue ActiveRecord::RecordNotFound
-        flash[:error] = Spree.t(:stock_locations_need_a_default_country)
-        redirect_to admin_stock_locations_path
+        unless @stock_location.country
```
Comment (dalizard) [inline backend/app/controllers/spree/admin/stock_locations_controller.rb]: @aishmita-aggarwal :thought_balloon: What's the reason behind this change? The [`Spree::Country.default`](https://github.com/spree/spree/blob/master/core/app/models/spree/country.rb#L18) method will raise an exception (always) if no record is found.

File: core/spec/models/spree/country_spec.rb [hunk 2]
```diff
@@ -10,9 +10,14 @@
...
     end
 
-    it 'will return the US if config is not set' do
+    it 'will return america if config is not set' do
       america.touch
       expect(described_class.default.id).to eql america.id
     end
+
+    it 'will return the first record if config is not set and america is not created' do
+      canada.touch
```
Comment (tanmay3011) [inline core/spec/models/spree/country_spec.rb]: @aishmita-aggarwal shouldn't this be done in before block?

File: backend/app/controllers/spree/admin/stock_locations_controller.rb (hunk 1 above)
Comment (tanmay3011) [inline backend/app/controllers/spree/admin/stock_locations_controller.rb]: @dalizard She has also changed the default method so now it won't raise an exception

Comment (aishmita-aggarwal) [inline backend/app/controllers/spree/admin/stock_locations_controller.rb]: @dalizard I believe there is no need to raise exceptions while accessing default country. Any thoughts?

**Comment by priyank-gupta** [comment]
@aishmita-aggarwal Please rebase with latest master.
:+1: 
@damianlegawiec @Mafi88 ?

**Comment by aishmita-aggarwal** [comment]
@priyank-gupta Done. :)

**Comment by bbonislawski** [comment]
I've rebased it and I'm going to merge it in #8585 . Thank you @aishmita-aggarwal for your great job!
